front/pages/w/[wId]/u/gens/index.tsx

```diff
@@ -696,14 +744,33 @@ export function TemplatesView({
                             }-500`,
                             instructions: newTemplateInstructions,
                           };
-                          console.log(new_template);
                           const curr_templates = templates.map((d) => d);
                           if (selectedTemplate == -1) {
+                            await fetch(`/api/w/${workspaceId}/templates`, {
```

Comment:
The API endpoint code is missing right?

Also I thiiiink we want to have an upsert logic here. Maybe we add a `sId` to GensTemplate that we can generate client side (see @philipperolet 's PR: https://github.com/dust-tt/dust/pull/685) for code to generate these ids client side